akka-distributed-data/src/test/scala/akka/cluster/ddata/ORMapSpec.scala

```diff
@@ -364,6 +364,56 @@ class ORMapSpec extends WordSpec with Matchers {
       merged3.entries("b").elements should be(Set("B2", "B3"))
     }
 
+    "have the ususal anomalies for remove+updated scenario" in {
+      // please note that the current ORMultiMap has the same anomaly
+      // because the condition of keeping global vvector is violated
+      // by removal of the whole entry for the removed key "b" which results in removal of it's value's vvector
+      val m1 = ORMap.empty.put(node1, "a", ORSet.empty.add(node1, "A")).put(node1, "b", ORSet.empty.add(node1, "B"))
+      val m2 = ORMap.empty.put(node2, "c", ORSet.empty.add(node2, "C"))
+
+      // m1 - node1 gets the update from m2
+      val merged1 = m1 merge m2
+      // m2 - node2 gets the update from m1
+      val merged2 = m2 merge m1
+
+      // RACE CONDITION!
+      val m3 = merged1.resetDelta.remove(node1, "b")
+      // let's imagine that m3 (node1) update gets propagated here (full state or delta - doesn't matter)
+      // and is in flight, but in the meantime, an element is being added somewhere else (m4 - node2)
+      // and the update is propagated before the update from node1 is merged
+      val m4 = merged2.resetDelta.updated(node2, "b", ORSet.empty[String])(_.add(node2, "B2"))
+      // and later merged on node1
+      val merged3 = m3 merge m4
+      // and the other way round...
+      val merged4 = m4 merge m3
+
+      // result -  the element "B" is kept on both sides...
+      merged3.entries("a").elements should be(Set("A"))
+      merged3.entries("b").elements should be(Set("B", "B2"))
+      merged3.entries("c").elements should be(Set("C"))
+
+      merged4.entries("a").elements should be(Set("A"))
+      merged4.entries("b").elements should be(Set("B", "B2"))
+      merged4.entries("c").elements should be(Set("C"))
+
+      // but if the timing was slightly different, so that the update from node1
+      // would get merged just before update on node2:
+      val merged5 = (m2 merge m3).resetDelta.updated(node2, "b", ORSet.empty[String])(_.add(node2, "B2"))
+      // the update propagated ... and merged on node1:
+      val merged6 = m3 merge merged5
+
+      // then the outcome is different... because the vvector of value("b") was lost...
+      merged5.entries("a").elements should be(Set("A"))
+      // this time it's different...
+      merged5.entries("b").elements should be(Set("B2"))
+      merged5.entries("c").elements should be(Set("C"))
+
+      merged6.entries("a").elements should be(Set("A"))
+      // this time it's different...
+      merged6.entries("b").elements should be(Set("B2"))
+      merged6.entries("c").elements should be(Set("C"))
```

Comment:
yes, but for the brand new `ORMultiMapWithDeltaDelta` this will be fixed... and will work consistent independent of timing